src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -2185,7 +2186,19 @@ HasOverlappingShardInterval(ShardInterval **shardIntervalArray,
 bool
 CitusHasBeenLoaded(void)
 {
-	if (!MetadataCache.extensionLoaded || creating_extension)
```

Comment:
We should probably call `AcceptInvalidationMessages()` here. To make sure we've processed any outstanding messages.

Reply:
I have reorganized this part of the code such that when `creating_extension == true`  for citus we do not change the cached value of `extensionLoadedState` but simply return false.

This change is to make the following scneario work:

> BEGIN;
> 
>SET client_min_messages TO ERROR; 
>SET search_path TO public;
>**CREATE EXTENSION citus;**                            
>create table l1 (a int unique);                                    
>SELECT create_reference_table('l1');
> 

When` creating_extension=true`, If we change the extensionCreatedState to false during  creating_extension, it stayed false after this line causing the subsequent `create_reference_table `to fail.

This might be due to that we invalidate cache during preprocessing phase of the citus utility hook when 
   `CREATE EXTENSION citus;`
runs.  And if we recreated the cache during creating_extension (we set extensionCreatedState to false), we do not get another cache invalidation message that would invalidate the cache.

`AcceptInvalidationMessages(); `

did not eliminate the issue. 

I opted for not changing the cache during` creating_extension` phase. 

This change should not have any impact on performance since

 `  if (creating_extension) `

used to run previously every time even when the cache is valid.